[DOC] Rename OTX to OpenVINO™ Training Extensions (#1797)
open-edge-platform/training_extensions: tests/test_suite/ARCHITECTURE.md

Note that each test action class MUST have the following properties:

> - it MUST be derived from the base class `BaseOpenVINO™ Training ExtensionsTestAction`;

Author: ```suggestion
- it MUST be derived from the base class `BaseOTXTestAction`;
```

It's constructor has declaration

```python
def __init__(self, action: BaseOpenVINO™ Training ExtensionsTestAction, stages_storage: OpenVINO™ Training ExtensionsTestStagesStorageInterface):
```

- The `action` parameter here is the instance of action that is wrapped.
  It is kept inside the `OTXTestStage` instance.
- The `stages_storage` here is an instance of a class that allows to get a stage by name, this will
  be a test case class that connects all the test stages between each other and keeps in its fields
  results of all test stages between tests

>   (all the test case classes are derived from OpenVINO™ Training ExtensionsTestStagesStorageInterface)

Reviewer: ```suggestion
  (all the test case classes are derived from OTXTestStagesStorageInterface)
```

The `stages_storage` instance is also kept inside `OTXTestStage`, it will be used to get for each
stage its dependencies.
Note that the abstract interface class `OTXTestStagesStorageInterface` has the only abstract method
`get_stage` with declaration

```python

> def get_stage(self, name: str) -> "OpenVINO™ Training ExtensionsTestStage":

Reviewer: ```suggestion
def get_stage(self, name: str) -> "OTXTestStage":
```

It's constructor has declaration

```python

> def __init__(self, action: BaseOpenVINO™ Training ExtensionsTestAction, stages_storage: OpenVINO™ Training ExtensionsTestStagesStorageInterface):

Author: ```suggestion
def __init__(self, action: BaseOTXTestAction, stages_storage: OTXTestStagesStorageInterface):
```

```python
def generate_otx_integration_test_case_class(

>     test_actions_classes: List[Type[BaseOpenVINO™ Training ExtensionsTestAction]],

Reviewer: ```suggestion
    test_actions_classes: List[Type[BaseOTXTestAction]],
```

  - call constructor of the current action as
    `cur_action = action_cls(**cur_params)`
  - wraps the current action with the class `OTXTestStage` as follows:

>     `cur_stage = OpenVINO™ Training ExtensionsTestStage(action=cur_action, stages_storage=self)`

Reviewer: ```suggestion
    `cur_stage = OTXTestStage(action=cur_action, stages_storage=self)`
```

The constructor of the class `OTXTestHelper` has the following declaration

```python

> def __init__(self, test_creation_parameters: OpenVINO™ Training ExtensionsTestCreationParametersInterface):

Reviewer: ```suggestion
def __init__(self, test_creation_parameters: OTXTestCreationParametersInterface):
```

All the methods of the interface class are abstract methods without parameters that return
structures making this connection.

> Example of such implementation is the class `DefaultOpenVINO™ Training ExtensionsTestCreationParametersInterface` that

Reviewer: ```suggestion
Example of such implementation is the class `DefaultOTXTestCreationParametersInterface` that
```

```python
@abstractmethod

> def test_case_class(self) -> Type[OpenVINO™ Training ExtensionsTestCaseInterface]:

Reviewer: ```suggestion
def test_case_class(self) -> Type[OTXTestCaseInterface]:
```

  `external/deep-object-reid/tests/test_otx_training.py`
  -- the latter is more interesting, since deep-object-reid algo backend is different w.r.t. the
  mmdetection algo backend, and we implemented the default test case parameter class

>   `DefaultOpenVINO™ Training ExtensionsTestCreationParametersInterface` mostly for mmdetection.

Reviewer: ```suggestion
  `DefaultOTXTestCreationParametersInterface` mostly for mmdetection.
```

- The test class should have a static field `helper` defined as follows:
  ```python

>   helper = OpenVINO™ Training ExtensionsTestHelper(<test parameters class>())

Reviewer: ```suggestion
  helper = OTXTestHelper(<test parameters class>())
```
